### ISSUE TITLE
`non-pep695-generic-class` (`UP046`) - false negative on old `TypeVar` with `default` argument

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#20656]
### Summary

```py
from typing import TypeVar, Generic

T = TypeVar("T", default=int)
U = TypeVar("U")


class Foo(Generic[T]): ... # no error
class Bar(Generic[U]): ... # error
```

https://play.ruff.rs/31611e2c-077c-4291-bdeb-dd663a0f43f7

this behavior is mentioned in the [known problems section](https://docs.astral.sh/ruff/rules/non-pep695-generic-class/#known-problems) but i assume this was written before this feature was widely supported by type checkers, and i couldn't find an existing issue for this.

### Version

0.13.2

**Comment by ntBre** [comment]
Yeah, this is still a [TODO](https://github.com/astral-sh/ruff/blob/bedfc6fd063e08f4d1e76ae8a780162404079298/crates/ruff_linter/src/rules/pyupgrade/rules/pep695/mod.rs#L321) from when I was first working on the rule. It's tracked in https://github.com/astral-sh/ruff/issues/15642 but only mentioned in the body, so it's probably a bit hard to find.

I'll make this a sub-issue rather than a duplicate for visibility. Contributions welcome!

We'll need to make this a preview change since the rule has been stabilized in the meantime.
